lib/gui/app/pages/main/controllers/flash.js

```diff
@@ -25,19 +26,88 @@ const exceptionReporter = require('../../../modules/exception-reporter')
 const imageWriter = require('../../../modules/image-writer')
 const path = require('path')
 const store = require('../../../../../shared/store')
+const constraints = require('../../../../../shared/drive-constraints')
+const availableDrives = require('../../../../../shared/models/available-drives')
 
 module.exports = function (
+  $q,
   $state,
   $timeout,
-  FlashErrorModalService
+  FlashErrorModalService,
+  WarningModalService,
+  DriveSelectorService
 ) {
+  /**
+   * @summary Spawn a confirmation warning modal
+   * @function
+   * @public
+   *
+   * @param {Array<String>} warningMessages - warning messages
+   * @returns {Promise} warning modal promise
+   *
+   * @example
+   * confirmationWarningModal([ 'Hello, World!' ])
+   */
+  const confirmationWarningModal = (warningMessages) => {
+    return WarningModalService.display({
+      confirmationLabel: 'Continue',
+      rejectionLabel: 'Change',
+      description: [
+        warningMessages.join('\n\n'),
+        'Are you sure you want to continue?'
+      ].join(' ')
+    })
+  }
+
+  /**
+   * @summary Display warning tailored to the warning of the current drives-image pair
+   * @function
+   * @public
+   *
+   * @param {Array<Object>} drives - list of drive objects
```

Comment:
The type doc here needs updating to `Array<String>` I believe

Reply:
It requires `Object` in order for the `constraints` predicates to work

Reply:
Oh, right - missed that